### ISSUE TITLE
DNN crop layer behavior when "offset" parameter is unspecified

### ISSUE THREAD

**Issue by drkoller** [opencv/opencv#10577]
##### System information (version)
- OpenCV => 3.4.0
- Operating System / Platform => Linux

##### Detailed description

In the recent versions of Caffe, the crop layer offset parameters default to 0 if unspecified, which crops to the upper-left corner. However, the OpenCV dnn module crop layer implementation does a "crop to center" operation instead if there are no offset parameters. This results in different behavior for OpenCV vs. Caffe when importing Caffe networks that use a crop layer with no offset parameter specified, such as:

```
layer {
    name: "myCropLayer"
    type: "Crop"
    bottom: "imageToCrop"
    bottom: "imageRef"
    top: "imageCropped"
}
```

**Comment by dkurt** [comment]
@drkoller, Thank you! Feel free to prepare a PR.

**Comment by kumar1202** [comment]
Hello, @drkoller  and @dkurt , I want to work on this issue. Kindly guide me about this issue .
What should be done to resolve this issue and what are the resources(file) used?

**Comment by dkurt** [comment]
@kumar1202, go ahead! A common pipeline in follows: install an origin [Caffe framework](http://caffe.berkeleyvision.org/). Write a simple `.prototxt` with a single input and a single layer `Crop` in way mentioned in the issue. Generate random input and use Caffe to obtain a target output. Then we need to modify OpenCV's crop layer to achieve exactly the same output for the same input. You may look at https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_caffe_importer.cpp and https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_layers.cpp for examples.

**Comment by kumar1202** [comment]
@dkurt Thanks for your response. I will keep you reported on this issue and soon prepare a PR.

**Comment by kumar1202** [comment]
@drkoller No problem. I will work on another issue then and thanks for informing.

**Comment by drkoller** [comment]
This issue has been addressed by PR #10591.

We should be wary for any users that may have relied on the prior behavior of the Crop layer (crop to center). I wish Caffe had an optional parameter to choose center cropping, as some other DNN frameworks do.
